### PR TITLE
Physicsworld

### PR DESCRIPTION
**Pull request by jared-paul**
### Description
Added a PhysicsWorld class to help contain all the objects needed for the PhysicsSimulator (PhysicsBall, PhysicsRobot, etc)

### Testing Done
Have to work with @MathewMacDougall to merge his simulator PR with this one and use the tests he created

### Resolved Issues
#1046 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/backend/simulation/physics/physics_world.cpp
Comment (garethellis0) [inline src/software/backend/simulation/physics/physics_world.cpp]: Could we change this name to `b2_world`? The reason I'm asking is that I would've guessed that the type of `physics_world` is `PhysicsWorld`, which confused me a bit below until I double checked it. This way it also doesn't match up with the `physics_field` and `physics_ball` below, which both actually _are_ `Physics*` types.

File: src/software/backend/simulation/physics/physics_world.h
Comment (garethellis0) [inline src/software/backend/simulation/physics/physics_world.h]: Perhaps we could call this something like `current_timestamp`? Just thinking that if I was writing it out `PhysicsWorld::physics_world_timestamp` feels a bit redundant. 

Just a suggestion, I don't feel super strongly about this, so totally your call.

**Review by garethellis0 — CHANGES_REQUESTED** [review]
Looking good! No major comments!

File: src/software/backend/simulation/physics/physics_world.cpp
Comment (MathewMacDougall) [inline src/software/backend/simulation/physics/physics_world.cpp:76]: super nit: remove extra blank line

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Just a super-nit, but otherwise this looks great! Nothing else to add beyond Gareth's comments

File: src/software/backend/simulation/physics/physics_world.cpp
Comment (jared-paul) [reply src/software/backend/simulation/physics/physics_world.cpp:76]: done

File: src/software/backend/simulation/physics/physics_world.h
Comment (jared-paul) [reply src/software/backend/simulation/physics/physics_world.h]: I agree, updated

File: src/software/backend/simulation/physics/physics_world.cpp
Comment (jared-paul) [reply src/software/backend/simulation/physics/physics_world.cpp]: done

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada:

**Review by jonathanlew — DISMISSED** [review]
LGTM! :+1:

**Comment by MathewMacDougall** [comment]
Looks like CI is failing due to firmware being built on master. Should pass once #1130 is merged. May need to manually trigger a new build afterwards. LGTM otherwise!

**Comment by garethellis0** [comment]
Just merged #1130, so re-running CI.

**Comment by MathewMacDougall** [comment]
Did it re-run? I don't see the option to trigger it from the browser. @jared-paul you might want to try pushing an "empty" commit, or amending your last commit and force-pushing to trick CI into running again.

**Comment by garethellis0** [comment]
You need to be logged into travis using github, but then you should be able to re-run (button in the top right of this image):
Screenshot from 2020-01-21 10-34-38

Just double checked and it did run this morning. @jared-paul, you'll need to merge `master` into your branch I think, that should fix it.

**Review by MathewMacDougall — APPROVED** [review]
:+1: :shipit: LGTM

**Review by garethellis0 — APPROVED** [review]

**Comment by MathewMacDougall** [comment]
@jared-paul :shipit:

**Comment by jared-paul** [comment]
:shipit:

**Pull request merged**
